lib/ruby_lsp/requests/goto_relevant_file.rb

```diff
@@ -0,0 +1,74 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # Goto Relevant File is a custom [LSP
+    # request](https://microsoft.github.io/language-server-protocol/specification#requestMessage)
+    # that navigates to the relevant file for the current document.
+    # Currently, it supports source code file <> test file navigation.
+    class GotoRelevantFile < Request
+      extend T::Sig
+
+      sig { params(path: String).void }
+      def initialize(path)
+        super()
+
+        @workspace_path = T.let(Dir.pwd, String)
+        @path = T.let(path.delete_prefix(@workspace_path), String)
+      end
+
+      sig { override.returns(T::Array[String]) }
+      def perform
+        find_relevant_paths(@path)
+      end
+
+      private
+
+      sig { params(path: String).returns(T::Array[String]) }
+      def find_relevant_paths(path)
+        workspace_path = Dir.pwd
+        relpath = path.delete_prefix(workspace_path)
+
+        candidate_paths = Dir.glob(File.join("**", relevant_filename_pattern(relpath)))
+        return [] if candidate_paths.empty?
+
+        find_most_similar_with_jacaard(relpath, candidate_paths).map { File.join(workspace_path, _1) }
+      end
+
+      sig { params(path: String).returns(String) }
+      def relevant_filename_pattern(path)
+        input_basename = File.basename(path, File.extname(path))
+
+        relevant_basename_pattern =
+          if input_basename.end_with?("test") || input_basename.end_with?("spec")
```

Comment:
If we're adding the file's extension name to `input_basename`, under what circumstances will it end with just `test` or `spec`? Can we have test cases for them?